app/scenes/right_menu_drawer/right_menu_drawer_item.js

```diff
@@ -23,18 +23,23 @@ export default class MainMenuItem extends React.Component {
         style: React.PropTypes.oneOfType([
             React.PropTypes.object,
             React.PropTypes.array
-        ])
+        ]),
+        shouldRender: React.PropTypes.bool
```

Comment:
@hmhealey We are already using `shouldRender` in this way in `UserProfileRow` & `ChannelInfoRow`. It was probably taken from there as an example for this PR. Maybe we can wait to change them there in a separate PR.
I'm all for seeking parity with the webapp when there's no downside, but to me, neither `shouldRender` nor `show` seems appropriate in this case. If the component will render by default, then I think it would be more intuitive to reverse the effect being implied by the property name.  e.g. `hide` with a default value of false.
Also words like `show` and `hide` could easily cause confusion about their usage. They have a stronger association with visibility manipulation than with rendering. Especially when in the context of drawers, modals, etc. `shouldRender` at least is more explicit.